src/dmd/backend/disasm86.d

```diff
@@ -1776,7 +1776,27 @@ void puts(const(char)* s)
     }
 }
 
-
+/**
+ * Holding area for functions that implement X86 instruction patterns
+ * It's a mixin template so it can be mixed into `disassemble` to access
+ * it's state.
+ */
+mixin template PatternFunctions()
+{
+    /**
+     * Implements `xmm1, xmm2/m128, imm8`
+     * Params:
+     *   indexOffset = this will be called from various amounts of
+     *                 lookahead into the code buffer which means an index is needed
+     *                 for finding the right offset to the register.
+     */
+    void xmm_xmm_imm8(uint indexOffset)
```

Comment:
Why this needs to be in a mixin? Only used one time and restrictive for this use case (where parameters are in the parent context scope).

Reply:
Because there will be more than one of them

Reply:
And look at how messy the place it is mixin-ed is